pkg/parser/workflows.go

```diff
@@ -246,12 +246,70 @@ func (doc *YamlDocument) parseContext(node *sitter.Node) []ast.TextAndRange {
 	return doc.getNodeTextArrayWithRange(node)
 }
 
-func (doc *YamlDocument) parseSingleJobRequires(node *sitter.Node) []ast.TextAndRange {
-	array := doc.getNodeTextArrayWithRange(node)
-	res := []ast.TextAndRange{}
-	for _, require := range array {
-		res = append(res, ast.TextAndRange{Text: require.Text, Range: require.Range})
+func (doc *YamlDocument) parseSingleJobRequires(requiresNode *sitter.Node) []ast.Require {
+	blockSequenceNode := GetChildSequence(requiresNode)
+	res := make([]ast.Require, 0)
```

Comment:
**Nit-picking:** Instead of initializing size & capacity to 0, we can pre-initialize, to avoid re-allocating on `append`?

As far as I understand the rest of the code, we don't know the size of the slice for certain, as we might [skip an invalid requires node](https://github.com/CircleCI-Public/circleci-yaml-language-server/pull/315/files#diff-07c13eb0887d9f118d1f533c768dc6f4da01cb1c01d7bf15054bb25874da68a0R289-R291). But what do you think of initializing to max capacity, as this will be a majority of the cases?

```suggestion
	res := make([]ast.Require, 0, requiresNode.ChildCount())
```

Not mandatory, but as parser performance can be critical to UX, it could be a nice touch. Would this work? Or am I misunderstanding something.

Reply:
Good call! That should work because `requiresNode.ChildCount()` should be the max capacity.